packages/headless/src/features/facets/category-facet-set/category-facet-set-slice.ts

```diff
@@ -81,10 +81,13 @@ export const categoryFacetSetReducer = createReducer(
         const pathToSelection = path.slice(0, path.length - 1);
 
         for (const segment of pathToSelection) {
-          const parent = activeLevel[0];
+          let parent = activeLevel[0];
+          const missingParent = !parent;
 
-          if (segment !== parent.value) {
-            return;
+          if (missingParent || segment !== parent.value) {
+            parent = buildCategoryFacetValueRequest(segment, retrieveCount);
+            activeLevel.length = 0;
+            activeLevel.push(parent);
```

Comment:
Is this meant to change the `children` of a category facet value?

If so, I would prefer not to mutate its children from a direct reference to its array.

Otherwise, I think we should replace this with:
```suggestion
            activeLevel = [parent];
```

Reply:
It is indeed a mutation, clearing any existing children, and pushing a new parent.
A direct assignment would not affect the reference, instead setting `activeLevel` variable to a new value, causing the reducer to not build the path correctly.